scapy/contrib/automotive/someip.py

```diff
@@ -129,23 +142,76 @@ class SOMEIP(Packet):
             RET_E_MALFORMED_MSG: "E_MALFORMED_MESSAGE",
             RET_E_WRONG_MESSAGE_TYPE: "E_WRONG_MESSAGE_TYPE",
         }),
+        ConditionalField(BitField("offset", 0, 28),
+                         lambda pkt: SOMEIP._is_tp(pkt)),
+        ConditionalField(BitField("res", 0, 3),
+                         lambda pkt: SOMEIP._is_tp(pkt)),
+        ConditionalField(BitField("more_seg", 0, 1),
+                         lambda pkt: SOMEIP._is_tp(pkt))
```

Comment:
What this means is that you could simply pass `SOMEIP._is_tp` instead of the lambda

Reply:
Yes I tried that, but apparently SOMEIP is not defined at that moment, because I get the following error:
```Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/buri/PycharmProjects/scapy/scapy/main.py", line 178, in load_contrib
    importlib.import_module("scapy.contrib." + name)
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/buri/PycharmProjects/scapy/scapy/contrib/automotive/someip.py", line 69, in <module>
    class SOMEIP(Packet):
  File "/home/buri/PycharmProjects/scapy/scapy/contrib/automotive/someip.py", line 145, in SOMEIP
    ConditionalField(BitField("offset", 0, 28), SOMEIP._is_tp),
NameError: name 'SOMEIP' is not defined
```

Reply:
If you move the function before, you might be able to use it without using `SOMEIP.`

Reply:
I moved it before but now I get this error:

```
>>> load_contrib("automotive.someip")
>>> s = SOMEIP()
>>> s.msg_type=0x21
>>> s.show()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/buri/PycharmProjects/scapy/scapy/packet.py", line 1169, in show
    return self._show_or_dump(dump, indent, lvl, label_lvl)
  File "/home/buri/PycharmProjects/scapy/scapy/packet.py", line 1125, in _show_or_dump
    if isinstance(f, ConditionalField) and not f._evalcond(self):
  File "/home/buri/PycharmProjects/scapy/scapy/fields.py", line 219, in _evalcond
    return self.cond(pkt)
TypeError: 'staticmethod' object is not callable
```

Reply:
I am really not sure how to do it in another way.
I think the lambda is necessary in this case.